src/context/category-context.js

```diff
@@ -0,0 +1,28 @@
+import axios from "axios";
+import React from "react"
+import { createContext, useContext, useEffect, useState } from "react";
+
+const CategoryContext=createContext();
+
+const CategoryProvider=({children})=>{
+const [category,setCategory]=useState([]);
+
+useEffect(()=>{
+    (async () => {
+        try {
+          const response = await axios.get("/api/categories");
+          setCategory(response.data.categories);
+        } catch (error) {
+          console.log(error);
```

Comment:
You may create a component for error to show errors if theres any. For now it's good.